### ISSUE TITLE
Chord not working when integrated terminal has focus

### ISSUE THREAD

**Issue by carlocardella** [Microsoft/vscode#23230]
Version 1.10.2
Shell 1.4.6
Renderer 53.0.2785.143
Node 6.5.0
Windows 10 Enterprise

I want to overwrite the keyboard hotkey to toggle the integrated terminal ( ctrl+` ): everything works fine as long as the editor has focus, while if the terminal has focus the hotkey works only if it's NOT a chord

**Steps to Reproduce**:
This works fine
```
{
        "key": "ctrl+t",
        "command": "workbench.action.terminal.toggleTerminal"
}
```

This works only when editorTextFocus (the keys don't matter, I tried with multiple combinations)
```
{
        "key": "ctrl+t ctrl+t",
        "command": "workbench.action.terminal.toggleTerminal"
}
```
When editorTextFocus the integrated terminal is displayed and has the focus; press the chord again and nothing happens. Give the text editor focus again, press the chord and the integrated terminal is toggled once again

**Comment by Tyriar** [comment]
Do you have anything bound to ctrl+t in addition to your ctrl+t ctrl+t keybinding?

**Comment by carlocardella** [comment]
These are the keybindings involving ctrl+t

//default keybindings
{ "key": "ctrl+k ctrl+t",         "command": "workbench.action.selectTheme" },
{ "key": "ctrl+t",                "command": "workbench.action.showAllSymbols" },

//my custom keybindings
{ "key": "ctrl+t ctrl+t", "command": "workbench.action.terminal.toggleTerminal" },

**Comment by Tyriar** [comment]
AFAIK the chord will never get hit because a ctrl+t keybinding is present. Try removing it (- the command):

```js
{ "key": "ctrl+t",                "command": "-workbench.action.showAllSymbols" },
```

**Comment by carlocardella** [comment]
Keybindings priority is from the bottom-up, right? Unfortunately this did not help:

    {
        "key": "ctrl+t ctrl+t",
        "command": "workbench.action.terminal.toggleTerminal"
    },
    {
        "key": "ctrl+t",
        "command": "-workbench.action.showAllSymbols"
    },

**Comment by Tyriar** [comment]
Something weird might be happening since there's a chord plus the command is both meant to skip and not skip the terminal (toggleTerminal is in `terminal.integrated.commandsToSkipShell` but showAllSymbols is not).

After removing showAllSymbols does toggleTerminal work fine outside of the terminal? If so, what happens when the terminal is focused? ctrl+t is a bash command that appears to swap the last 2 characters and move the cursor 1 to the right, is that happening if you have something in the prompt?

**Comment by carlocardella** [comment]
Outside the terminal chords always work fine. I tried to use a different chord (e.g. "ctrl+o ctrl+t") with the same behavior as described initially. 
When focus is on the terminal, if I press the chord I just get the pre-Win10 cmd behavior when ctrl is pressed, like this: 


```
Microsoft Windows [Version 10.0.14393]
(c) 2016 Microsoft Corporation. All rights reserved.

C:\Git\>$psversiontable
'$psversiontable' is not recognized as an internal or external command,
operable program or batch file.


C:\Git\>^O^T
```

**Comment by Tyriar** [comment]
Actually I know why this is happening, here's the code responsible https://github.com/Microsoft/vscode/blob/2a823213ce8c13efb457bd271dfd2d30ba6499c6/src/vs/workbench/parts/terminal/electron-browser/terminalInstance.ts#L234

```ts
			// Skip processing by xterm.js of keyboard events that resolve to commands described
			// within commandsToSkipShell
			const standardKeyboardEvent = new StandardKeyboardEvent(event);
			const resolveResult = this._keybindingService.softDispatch(standardKeyboardEvent, standardKeyboardEvent.target);
			if (resolveResult && this._skipTerminalCommands.some(k => k === resolveResult.commandId)) {
				event.preventDefault();
				return false;
			}
```

Since there is no non-chord command that matches `ctrl+t` it will not skip the terminal. We made a design decision earlier on to break chord keybindings within the terminal in order to support ctrl/cmd+k to clear the terminal, and because ctrl+k is a keybinding in bash. Because of this, the `softDispatch` call which asks the keybinding service what command the keybinding will do if the key is send there, then the list is checked against the `commandsToSkipShell` setting. Chords are explicitly ignored here however as it's hard to determine whether a chord should skip the terminal or not.

TLDR it's a bit complicated and I don't think we'll end up supporting this, at least not before we work out the kinks in the `commandsToSkipShell` setting https://github.com/Microsoft/vscode/issues/19770
